src/cpu/aarch64/acl_gemm_convolution.hpp

```diff
@@ -0,0 +1,194 @@
+/*******************************************************************************
+* Copyright 2020 Arm Ltd. and affiliates
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#ifndef CPU_AARCH64_ACL_GEMM_CONVOLUTION_HPP
+#define CPU_AARCH64_ACL_GEMM_CONVOLUTION_HPP
+
+#include "common/c_types_map.hpp"
+#include "common/memory_tracking.hpp"
+#include "common/primitive.hpp"
+
+#include "cpu/aarch64/acl_gemm_convolution_utils.hpp"
+#include "cpu/gemm/gemm.hpp"
+
+#include "cpu/cpu_convolution_pd.hpp"
+
+#include "arm_compute/runtime/NEON/NEFunctions.h"
+#include "arm_compute/runtime/Scheduler.h"
+
+namespace dnnl {
+namespace impl {
+namespace cpu {
+namespace aarch64 {
+
+struct acl_obj_t {
+    arm_compute::NEGEMMConvolutionLayer gemm_conv;
+    arm_compute::Tensor src_tensor;
+    arm_compute::Tensor wei_tensor;
+    arm_compute::Tensor bia_tensor;
+    void *bia_memory;
```

Comment:
In general, for memory objects (like bias here) we use the [scratchpad](https://oneapi-src.github.io/oneDNN/dev_guide_attributes_scratchpad.html) mechanism. Resources can be used for this purpose, but this not an idiomatic way, as user cannot track these allocations that could be comparable with input/output data.

Since it should be relatively easy to fix, I would like us to do so before the release.

An example of using scratchpad:
- Booking in `pd_t::init()`: https://github.com/oneapi-src/oneDNN/blob/613d2308ed9041b01e9c3cbbd2a974bbec906d38/src/cpu/simple_sum.hpp#L108-L111
- Getting the pointer at execution: https://github.com/oneapi-src/oneDNN/blob/613d2308ed9041b01e9c3cbbd2a974bbec906d38/src/cpu/simple_sum.cpp#L54-L55

Reply:
Done. Not entirely sure about the 'key' I used however.

Reply:
`memory_tracking::names::key_none` is not very descriptive, but should work fine as long as you don't call/interact with other primitives within a given one. This is now the case -- convolution is always standalone, it doesn't rely or call anything else.

In future, you could consider adding `memory_tracking::names::key_conv_bia` or reuse existing one, say [`key_conv_padded_bias`](https://github.com/oneapi-src/oneDNN/blob/f6615f305b6efa2bf2d2f573ac7e4760bbbd0d3e/src/common/memory_tracking.hpp#L176) (which is not entirely correct, but still reflect that buffer belongs to convolution bias). Or, maybe the problem will go away, and you want need the scratchpad at all. Anyways, thanks for the fix. Resolving the discussion.